src/scalper.ts

```diff
@@ -677,28 +678,27 @@ Spot Δ: ${spotDelta} Offset Δ ${this.deltaOffset} Fair Value: ${fairValue}`,
           for (const jupTx of [setupTransaction, swapTransaction, cleanupTransaction].filter(
             Boolean,
           )) {
-            if (!jupTx) {
-              continue;
-            }
             const txid = await sendAndConfirmTransaction(this.connection, jupTx, [this.owner]);
-            if (swapTransaction) {
+            if (jupTx === swapTransaction) {
               const spotDeltaUpdate = jupValues.qty;
               hedgeDeltaTotal += spotDeltaUpdate;
               console.log(this.symbol, 'Jupiter Hedge via', jupValues.venue, 'Price', jupValues.price, 'Qty', jupValues.qty, `https://solana.fm/tx/${txid}${cluster?.includes('devnet') ? '?cluster=devnet' : ''}`);
             }
           }
-          if (Math.abs(hedgeDeltaTotal * fairValue) < (deltaThreshold * fairValue)) {
-            console.log(this.symbol, 'Delta Netural: Jupiter Hedge');
-            return;
-          }
           console.log(this.symbol, 'Adjust', hedgeSide, 'Price', hedgePrice, 'to', jupValues.price, 'Remaining Qty', hedgeDeltaTotal);
           hedgePrice = jupValues.price;
         } else {
           console.log(this.symbol, 'No Jupiter Route Found Better than', hedgePrice);
         }
         hedgeDeltaClip = hedgeDeltaTotal / spliceFactor;
       } else {
-        console.log(this.symbol, 'Sufficient liquidity. Sweep OpenBook', spliceFactor);
+        console.log(this.symbol, 'Sufficient liquidity. Sweep OpenBook');
+      }
+
+      // Return early if jupiter sweeping got within the threshold.
+      if (Math.abs(hedgeDeltaTotal) < deltaThreshold) {
```

Comment:
Moved down in preparation for taking out the jupiter sweep logic into a new function. Should not be functionally different